proseco/guide_report.py

```diff
@@ -0,0 +1,290 @@
+    out_html = template.render(context)
+    out_filename = obsdir / 'guide_index.html'
+    with open(out_filename, 'w') as fh:
+        fh.write(out_html)
+
+
+def make_cand_report(guides, cand_guides, context, obsdir):
+
+    n_stages = np.max(cand_guides['stage'])
+    context['cand_guides'] = []
+    for ii, guide in enumerate(cand_guides):
+        print('Doing detail for star {}'.format(guide['id']))
+        select = 'SELECTED' if guide['id'] in guides['id'] else 'not selected'
+
+        # Add debugging information if the star was in include/exclude lists
+        if (guide['id'] in guides.include_ids_guide) or (guide['id'] in guides.exclude_ids_guide):
+            select = select + ' - forced with include/exclude'
+        rep = {
+            'id': guide['id'],
+            'selected': select
+        }
+        log = reject_info_to_report(guide['id'], guides.reject_info)
+        if guide['stage'] != -1:
+            if guide['stage'] == 0:
+                log.append(f"FORCE Selected in stage 0")
+            else:
+                log.append(f"Selected in stage {guide['stage']}")
+            if guide['id'] not in guides['id']:
+                log.append(f"Did not make final cut of {len(guides)} stars")
+        else:
+            log.append(f"Not selected in any stage ({n_stages} stages run)")
+        rep['log'] = log
+        guide_table = cand_guides[COLS][ii:ii + 1].copy()
+        guide_table['id'] = ['<a href="http://kadi.cfa.harvard.edu/star_hist/?agasc_id={0}" '
+                             'target="_blank">{0}</a>'
+                             .format(g['id']) for g in guide_table]
+        rep['guide_table'] = table_to_html(guide_table)
+
+        # Make the star detail plot
+        basename = f'guide_candidate_{guide["id"]}.png'
+        filename = obsdir / basename
+        rep['candidate_plot'] = basename
+        if not filename.exists():
+            plot(guide, guides, filename=filename)
+        context['cand_guides'].append(rep)
+
+
+def reject_info_to_report(starid, reject_info):
+    """
+    For a given agasc_id, get all of the related "reject" info in an array
+    """
+    log = []
+    for entry in reject_info:
+        if entry['id'] != starid:
+            continue
+        log.append(f"Not selected stage {entry['stage']}: {entry['text']}")
+    return log
+
+
+def plot(guide, cand_guides, filename=None):
+    """
+    For a given candidate guide, make a plot of with a one subplot of possible spoiler stars
+    and another subplot of the region of the dark map that would be dithered over.
+    """
+    # NOTE: all coordinates in plot funcs are "edge" coordinates.  Indexing an
+    # ACAImage pixel using `.aca` coordinates implies using the row, col
+    # coordinate of the lower-left corner of that pixel.
+
+    fig = plt.figure(figsize=(8, 4))
+
+    # First plot: star spoilers
+    ax = fig.add_subplot(1, 2, 1)
+    plot_spoilers(guide, cand_guides, ax)
```

Comment:
Though since this just uses the table of the other candidates to get cand_guides.stars, it would make more sense to just pass cand_guides.stars .

Reply:
Doesn't hurt to have the whole thing and makes the API more consistent.  These may end up as methods anyway in the next round of refactor that I'm planning.